java/ql/src/semmle/code/java/frameworks/android/Android.qll

```diff
@@ -33,18 +33,39 @@ class AndroidComponent extends Class {
 /** An Android activity. */
 class AndroidActivity extends AndroidComponent {
   AndroidActivity() { this.getASupertype*().hasQualifiedName("android.app", "Activity") }
+
+  /** Holds if this Android component is configured as `exported` or has intent filters configured without `exported` explicitly disabled in an `AndroidManifest.xml` file. */
+  override predicate isExported() {
```

Comment:
Let's introduce a common supertype for these three classes (`ExportableAndroidComponent` perhaps) so that we don't duplicate this code 3 times

Reply:
Added <code>ExportableAndroidComponent</code>, which helps to remove duplicate code.